packages/react-tabs/src/components/Tab/useTab.ts

```diff
@@ -15,24 +15,33 @@ import { SelectTabEvent } from '../TabList/TabList.types';
  * @param ref - reference to root HTMLElement of Tab
  */
 export const useTab_unstable = (props: TabProps, ref: React.Ref<HTMLElement>): TabState => {
-  const { content, icon, value } = props;
+  const { content, disabled: tabDisabled = false, icon, value } = props;
 
-  const { appearance, selected, onRegister, onUnregister, onSelect, size, vertical } = useContextSelector(
-    TabListContext,
-    ctx => ({
-      appearance: ctx.appearance,
-      selected: ctx.selectedValue === value,
-      onRegister: ctx.onRegister,
-      onUnregister: ctx.onUnregister,
-      onSelect: ctx.onSelect,
-      size: ctx.size,
-      vertical: !!ctx.vertical,
-    }),
-  );
+  const {
```

Comment:
The objective of using context selectors is to avoid re-renders if the context value changes. Internally `useContextSelector` uses a shallow compare for the result of the selector and then determines if a re-render should be forced.

Here, the selector you are using creates a new object that will result in rerenders every time any value in your context changes.

Using individual context selectors would make more sense here because primitive values can be shallow compared. If any of the individual selectors return new results, this would still result in a rerender. However, this is better than the current situation wher,e if any context values not being used changes, it will cause a rerender.

```ts
// context value
interface TabListContext {
  appearance: boolean;
  disabled: boolean;
  foo: string;
}

// Good- if foo changes then your component does not rerender
const appearance = useContextSelector(TabListContext, ctx => ctx.appearance)
const listDisabled = useContextSelector(TabListContext, ctx => ctx.disabled)

// Bad - if foo changes then your component rerenders (even if it does not use foo
const { appearance, listDisabled } = useContextSelector(TabListContext, ctx => ({ appearance: ctx.appearance, disabled: ctx.disabled })
```

Reply:
I didn't know that! Thanks! Will update.